addons/textalog/nodes/dialogue_base_node/portraits_node/portraits_node.gd

```diff
@@ -40,19 +40,25 @@ func add_portrait(
 		emit_signal("portrait_added", character, portrait)
 		return
 	
-	# Remove previous node
-	if character in portraits:
-		remove_portrait(character)
-		emit_signal("portrait_changed", character, portrait)
 	
-	var _texture_rect:TextureRect = TextureRect.new()
-	connect("tree_exiting", _texture_rect, "queue_free")
+	var _texture_rect:TextureRect
+	var changed: bool = false
+
+	# Use previous node as reference
+	if character in portraits:
+		_texture_rect = portraits.get(character, null)
+		changed = true
+	# Create a new node as no previous node exists
+	if not is_instance_valid(_texture_rect):
+		_texture_rect = TextureRect.new()
+		portraits[character] = _texture_rect
```

Comment:
There's a problem here, the node is going to be added on 
https://github.com/AnidemDex/Godot-DialogPlugin/blob/1a68a785d76e704eae748ac0842c25000f51d9e0/addons/textalog/nodes/dialogue_base_node/portraits_node/portraits_node.gd#L66

Adding a node again can raise an error, since it's already on tree.